Enforce write privileges for token writes

Mats-SX opened the pull request:
Cypher was recently changed to not perform token writes in read queries.
This commit adds a test that will fail if it would do so. It further enforces that
acquiring `TokenWriteOperations` requires the `AccessMode` to allow writes.

This PR adds a test for the change in #8100.

Mats-SX committed:
Enforce write privileges for token writes

Cypher was recently changed to not perform token writes in read queries.
This commit adds a test that will fail if it would do so. It further enforces that
acquiring `TokenWriteOperations` requires the `AccessMode` to allow writes.

lutovich commented on community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java:
```diff
@@ -108,6 +108,12 @@ public ProcedureCallOperations procedureCallOperations()
     {
         accessCapability.assertCanWrite();
+
+        if ( !transaction.mode().allowsWrites() )
```
Similar checks already exist in `#dataWriteOperations()` and `#schemaWriteOperations()`. I think it is worth to create a method like `assertWritesAllowed(String message)` at this point.

lutovich reviewed (changes requested):
@Mats-SX change looks good to me. It is possible to prettify code a bit and extract an assertion method. More importantly, build has test failures related to the change.

OliviaYtterbrink committed:
Extract private helper for access checks

OliviaYtterbrink committed:
Remove token write from read-only test procedures

Mats-SX commented:
@lutovich Thanks for the review! We made this commit to address your comment: https://github.com/neo4j/neo4j/pull/8113/commits/c0c061de74e884838608f757a3db4934e0b67376

Mats-SX committed:
Explicitly check for schema write privileges

This overcomes an issue where the system would try to acquire
`TokenWriteOperations` before `SchemaWriteOperations`, resulting
in a suboptimal error message.

Mats-SX committed:
Remove unused code
